packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -115,6 +117,27 @@ export default class Editor implements IEditor {
             trustedHTMLHandler: options.trustedHTMLHandler || ((html: string) => html),
             zoomScale: zoomScale,
             sizeTransformer: options.sizeTransformer || ((size: number) => size / zoomScale),
+            getVisibleViewport:
+                options.getVisibleViewport ||
+                (() => {

```

Comment:
Suggest put this function into roosterjs-editor-dom and export it, then when you override it, you can still call sthis function and make minor change to the result, but no need to override the whole function.

You can pass an array of elements and calculate the intersect area for all of them.

Question: if there is no intersect area, do we want to return an invalid react (e.g. top is larger than bottom), or just return null?

Reply:
Added the new function, I also added an additional parameter for Rects array, I think this would be useful in case we need to calculate or modify rect outside of the function.